src/main/java/in/projecteka/consentmanager/clients/DiscoveryServiceClient.java

```diff
@@ -34,4 +36,21 @@
                 .flatMap(responseSpec -> responseSpec.bodyToMono(PatientResponse.class));
     }
 
+    public Mono<Boolean> requestPatientFor(PatientRequest request, String url, String hipId) {
+        return tokenGenerator.get()
+                .map(token ->
+                        webClientBuilder.build()
+                                .post()
+                                .uri(url + PATIENTS_CARE_CONTEXTS_DISCOVERY_URL_PATH)
+                                .header(AUTHORIZATION, token)
+                                .header("X-HIP-ID", hipId)
```

Comment:
Should cm-id header also be sent here?

Reply:
no. Its CM who is sending the request and it just adds the targeted the HIP ID